### PR TITLE
Add inception v1 training example

### PR DESCRIPTION
**Pull request by yangw1234**
todos:
- [x] add readme
- [x] add liscense header

### CODE REVIEW

File: pyzoo/dev/check-license [hunk 1]
```diff
@@ -23,7 +23,7 @@ header="Copyright 2018 Analytics Zoo Authors"
 
 echo "Checking Analytics Zoo license for Python files..."
 
-find $DL_PYTHON_HOME ! -name "pep8-1.7.0.py" ! -name "tf_graph_util.py" -name "*.py" -print0 | while read -d $'\0' file
+find $DL_PYTHON_HOME ! -name "pep8-1.7.0.py" ! -name "tf_graph_util.py" -name "*.py" | grep -v "tfpark/inception/nets/" | while read -d $'\n' file
```
Comment (yangw1234) [inline pyzoo/dev/check-license:26]: Hi @hkvision, could you help review this change?

Comment (hkvision) [reply pyzoo/dev/check-license:26]: The change LGTM.
Could you please add a comment in this file saying something like "Excluding several TensorFlow releated files since they are migrated from TensorFlow"?

Comment (yangw1234) [reply pyzoo/dev/check-license:26]: sure

**Comment by yangw1234** [comment]
Jenkins passed http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/3156/

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+# Inception Model on Imagenet
+This example demonstrates how to use Analytics-zoo to train a TensorFlow [Inception v1](https://arxiv
```
Comment (Wesley-Du) [inline pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:2]: Seems there is a blank space and breaks the URL: [Inception v1](https://arxiv .org/abs/1409.4842)

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+.org/abs/1409.4842) model on the [ImageNet](http://image-net.org/index) data.
+## Get the JAR
+You can build one by refer to the
+[Build Page](https://analytics-zoo.github.io/master/#ScalaUserGuide/install/#download-analytics-zoo-source) from the source code. We
+will release a pre-build package soon.
+
+## Prepare the data
+You can download imagenet-2012 data from <http://image-net.org/download-images> and put them in this directory.
```
Comment (Wesley-Du) [inline pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:10]: Specify the "this directory"?

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+bash prepare_data.sh $parition_num
+```
+
+This command will generate the hadoop sequence files in the `sequence` folder.
+
+Then you can put the sequence files to hdfs. E.g.
+
+```bash
+hadoop fs -mkdir -p /user/root
+hadoop fs -copyFromLocal sequence/ /user/root/
```
Comment (Wesley-Du) [inline pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:28]: Is the "sequence" folder with format like hdfs://[IP/host name]:/user_folder? Xin told me it's this format.

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+[Build Page](https://analytics-zoo.github.io/master/#ScalaUserGuide/install/#download-analytics-zoo-source) from the source code. We
+will release a pre-build package soon.
+
+## Prepare the data
+You can download imagenet-2012 data from <http://image-net.org/download-images> and put them in this directory.
+ 
+After you download the files(**ILSVRC2012_img_train.tar** and **ILSVRC2012_img_val.tar**), 
+run the following commands to prepare the data.
+
+The first arguments of `prepare_data.sh` is the output partition number and it is recommended to be the total
```
Comment (Wesley-Du) [inline pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:15]: Should set $partition_num to 28 or 224 if use the spark standalone example below?

--executor-cores 28 \  
--total-executor-cores 224 \

Comment (yangw1234) [reply pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:15]: it should be 224, right @qiuxin2012

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md (hunk 4 above)
Comment (yangw1234) [reply pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:28]: synced offline, current way is fine

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md (hunk 2 above)
Comment (yangw1234) [reply pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md]: fixed

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md (hunk 3 above)
Comment (yangw1234) [reply pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md]: fixed

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+--maxIteration 62000
+```
+
+* Spark yarn client mode, example command
+```bash
+export SPARK_HOME=the root directory of Spark
+export ANALYTICS_ZOO_HOME=the dist directory under the Analytics Zoo project
+# You can uncomment the following line if hyper-threading is disabled in your cluster
+# export KMP_AFFINITY=granularity=fine,verbose,compact
+DATA_PATH=hdfs://path/to/sequence/files
```
Comment (qiuxin2012) [inline pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:60]: hdfs://[IP/hostname:port]/path/to/sequence

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+## Train the Model
+* Spark standalone, example command
+```bash
+export SPARK_HOME=the root directory of Spark
+export ANALYTICS_ZOO_HOME=the dist directory under the Analytics Zoo project
+# You can uncomment the following line if hyper-threading is disabled in your cluster
+# export KMP_AFFINITY=granularity=fine,verbose,compact
+DATA_PATH=hdfs://path/to/sequence/files
```
Comment (qiuxin2012) [inline pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:38]: hdfs://[IP:port]/path/to/sequence

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+${ANALYTICS_ZOO_HOME}/bin/spark-submit-python-with-zoo.sh \
+--master yarn \
+--deploy-mode client \
+--executor-cores 28 \
```
Comment (qiuxin2012) [inline pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:65]: 56 and 4?

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+```bash
+export SPARK_HOME=the root directory of Spark
+export ANALYTICS_ZOO_HOME=the dist directory under the Analytics Zoo project
+# You can uncomment the following line if hyper-threading is disabled in your cluster
+# export KMP_AFFINITY=granularity=fine,verbose,compact
+DATA_PATH=hdfs://path/to/sequence/files
+
+${ANALYTICS_ZOO_HOME}/bin/spark-submit-python-with-zoo.sh \
+--master spark://xxx.xxx.xxx.xxx:xxxx \  
+--executor-cores 28 \
```
Comment (qiuxin2012) [inline pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:42]: 56?

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+--deploy-mode client \
+--executor-cores 28 \
+--num-executors 8 \ 
+--executor-memory 175G \ 
+--driver-memory 20G \ 
+--conf spark.network.timeout=10000000  inception.py \ 
+--batchSize 1792 \
+--learningRate 0.0896 \
+-f $DATA_PATH \
+--checkpoint /tmp/models/inception \
```
Comment (qiuxin2012) [inline pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:73]: mkdir first?

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md (hunk 6 above)
Comment (yangw1234) [reply pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md]: Fixed.

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md (hunk 7 above)
Comment (yangw1234) [reply pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md]: Fixed

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md (hunk 10 above)
Comment (yangw1234) [reply pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:73]: Fixed

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md (hunk 9 above)
Comment (yangw1234) [reply pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md]: Fixed

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md (hunk 8 above)
Comment (yangw1234) [reply pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md]: Fixed

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md (hunk 5 above)
Comment (qiuxin2012) [reply pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:14]: It's just a number for sequence files. Number of partitions, when you use spark to read this sequence files from HDFS, will dependens on blocksize(usually 256MB) in HDFS's configuration. The total number of partitions  is `totalsize / 256MB`.
So, here, we should use the number of available cores in the machine who execute this script.

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+```
+
+* Spark yarn client mode, example command
+```bash
+export SPARK_HOME=the root directory of Spark
+export ANALYTICS_ZOO_HOME=the dist directory under the Analytics Zoo project
+# You can uncomment the following line if hyper-threading is disabled in your cluster
+# export KMP_AFFINITY=granularity=fine,verbose,compact
+DATA_PATH=hdfs://[IP:port]/path/to/sequence/files
+mkdir -p /tmp/models/
```
Comment (qiuxin2012) [inline pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:63]: I think is `mkdir -p /tmp/models/inception`

Comment (yangw1234) [reply pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md:63]: checkpoint argument is used in 
`saver.save(optimizer.sess, options.checkpoint)`
And saver.save("/tmp/models/inception") will create checkpoint files in `/tmp/models/` with prefix `inception`

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md (hunk 5 above)
Comment (yangw1234) [reply pyzoo/zoo/examples/tensorflow/tfpark/inception/README.md]: fixed

**Comment by yangw1234** [comment]
I will first merge it. If any issues come up, I'll open another PR to fix it.

**Comment by qiuxin2012** [comment]
@yangw1234 I think we need some readme to tell user how to resume training.

File: pyzoo/zoo/examples/tensorflow/tfpark/inception/inception.py [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+        end_trigger = MaxIteration(options.maxIteration)
+
+    optimizer = TFOptimizer.from_loss(loss, optim,
+                                      val_outputs=[logits],
+                                      val_labels=[zero_based_label],
+                                      val_method=[Accuracy(), Top5Accuracy(), Loss()],
+                                      tensor_with_value={is_training: [True, False]},
+                                      model_dir="/tmp/logs")
+
+    optimizer.optimize(end_trigger=end_trigger, checkpoint_trigger=checkpoint_trigger)
```
[the remark this answers is not in the document]
Comment (qiuxin2012) [reply pyzoo/zoo/examples/tensorflow/tfpark/inception/inception.py:159]: @yangw1234
